modules/core/utils/upgrade_nomad_clients.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/bash
+set -euo pipefail
+
+read -p 'AWS_PROFILE: ' awsProfile
+read -p 'NOMAD_ADDR: ' nomadAddr
+read -p 'NOMAD_TOKEN: ' nomadToken
+read -p 'ASGName: ' ASGName
+
+export AWS_PROFILE=$awsProfile
+export NOMAD_ADDR=$nomadAddr
+export NOMAD_TOKEN=$nomadToken
+
+echo 'Getting old instance ID'
+aws autoscaling describe-auto-scaling-groups \
+    --auto-scaling-group-name $ASGName \
+    | jq --raw-output '.AutoScalingGroups[0].Instances[].InstanceId' \
+    | tee instance-ids.txt
```

Comment:
Let the user choose the directory to write the files to. Can default to `./`.

The entire script is basically working off this list of instances. In the case of some errors along the way, the user might have to restart the entire process. So a few thoughts:

1. Add an optional argument to allow the user to specify the list of instances instead of discovering them
1. Add an optional argument to not increase the size of the ASG if the user wants to